app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/actions/index.js

```diff
@@ -217,14 +310,62 @@ export const validateRule = (rule, dispatch, appMode) => {
                 error: "invalid script code",
               };
             }
+            /* NOT APPLYING LOGICAL VALIDATIONS FOR NOW */
+            // else {
+            //   const res = checkForLogicalErrorsInCode(script.value, script);
+            //   if (!res.isValid) {
+            //     if (res.error === scriptLogicalErrors.CONTAINS_HTML_CODE) {
+            //       output = {
+            //         result: false,
+            //         message: `Please remove unnecessary HTML tags from the code`,
+            //         error: "invalid script code",
+            //       };
+            //     }
+            //     else if (res.error === scriptLogicalErrors.DOM_LOAD_EVENT_LISTENER_AFTER_PAGE_LOAD) {
+            //       // dispatch(
+            //       //   actions.updateRulePairAtGivenPath({
+            //       //     pairIndex,
+            //       //     triggerUnsavedChangesIndication: true,
+            //       //     updates: {
+            //       //       [`scripts[${scriptIndex}].loadTime`]: GLOBAL_CONSTANTS.SCRIPT_LOAD_TIME.BEFORE_PAGE_LOAD,
+            //       //     },
+            //       //   })
+            //       // );
+
+            //       // just show a toast!!!
+
+            //       output = {
+            //         result: false,
+            //         message: `Updated script load time to before page load`,
+            //         error: "invalid script code",
+            //         editorToast: {
+            //           message: `Contains a DomContentLoaded listener being injected after page load`,
+            //           id: pair.id,
+            //           type: "info",
+            //           onlyInEditor: false,
+            //         },
+            //         ruleUpdated: true,
+            //       };
+            //     }
+            //   }
+            // }
           }
           //Check if URL isn't empty. Can be absolute or relative
-          else if (script.type === "url" && isEmpty(script.value)) {
-            output = {
-              result: false,
-              message: `Please enter a valid script URL`,
-              error: "invalid script url",
-            };
+          else if (script.type === "url") {
+            if (isEmpty(script.value)) {
+              output = {
+                result: false,
+                message: `Please enter a valid script URL`,
+                error: "invalid script url",
+              };
+            }
+            if (!isValidUrl(script.value)) {
+              output = {
+                result: false,
+                message: `Please enter a valid script URL`,
+                error: "invalid script url",
+              };
```

Comment:
If the error is same in both the cases then we can merge them with `||` operator. No?